backend/src/apiserver/storage/pipeline_store.go

```diff
@@ -425,3 +447,350 @@ func (s *PipelineStore) UpdatePipelineAndVersionsStatus(id string, status model.
+				"CreatedAtInSec": newPipelineVersion.CreatedAtInSec,
+				"Name":           newPipelineVersion.Name,
+				"Parameters":     newPipelineVersion.Parameters,
+				"PipelineId":     newPipelineVersion.PipelineId,
+				"Status":         string(newPipelineVersion.Status),
+				"CodeSourceUrl":  ""}).
+		ToSql()
+	if versionErr != nil {
+		return nil, util.NewInternalServerError(
+			versionErr,
+			"Failed to create query to insert version to pipeline version table: %v",
+			versionErr.Error())
+	}
+	pipelineSql, pipelineArgs, pipelineErr := sq.
+		Update("pipelines").
+		SetMap(sq.Eq{"DefaultVersionId": newPipelineVersion.UUID}).
+		Where(sq.Eq{"UUID": newPipelineVersion.PipelineId}).
+		ToSql()
+	if pipelineErr != nil {
+		return nil, util.NewInternalServerError(
+			pipelineErr,
+			"Failed to create query to update pipeline default version id: %v",
+			pipelineErr.Error())
+	}
+
+	// In a single transaction, insert new version and update default version.
+	tx, err := s.db.Begin()
+	if err != nil {
+		return nil, util.NewInternalServerError(
+			err,
+			"Failed to start a transaction: %v",
+			err.Error())
+	}
+
+	_, err = tx.Exec(versionSql, versionArgs...)
+	if err != nil {
+		tx.Rollback()
+		if s.db.IsDuplicateError(err) {
+			return nil, util.NewInvalidInputError(
+				"Failed to create a new pipeline version. The name %v already exist. Please specify a new name.", v.Name)
+		}
+		return nil, util.NewInternalServerError(err, "Failed to add version to pipeline version table: %v",
+			err.Error())
+	}
+	_, err = tx.Exec(pipelineSql, pipelineArgs...)
+	if err != nil {
+		tx.Rollback()
+		return nil, util.NewInternalServerError(err, "Failed to update pipeline default version id: %v",
+			err.Error())
+	}
+	if err := tx.Commit(); err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to create new pipeline version: %v",
+			err.Error())
+	}
+
+	return &newPipelineVersion, nil
+}
+
+func (s *PipelineStore) UpdatePipelineDefaultVersion(pipelineId string, versionId string) error {
+	sql, args, err := sq.
+		Update("pipelines").
+		SetMap(sq.Eq{"DefaultVersionId": versionId}).
+		Where(sq.Eq{"UUID": pipelineId}).
+		ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err, "Failed to create query to update the pipeline default version: %s", err.Error())
+	}
+	_, err = s.db.Exec(sql, args...)
+	if err != nil {
+		return util.NewInternalServerError(err, "Failed to update the pipeline default version: %s", err.Error())
+	}
+
+	return nil
+}
+
+func (s *PipelineStore) GetPipelineVersion(versionId string) (*model.PipelineVersion, error) {
+	return s.GetPipelineVersionWithStatus(versionId, model.PipelineVersionReady)
+}
+
+func (s *PipelineStore) GetPipelineVersionWithStatus(versionId string, status model.PipelineVersionStatus) (*model.PipelineVersion, error) {
+	sql, args, err := sq.
+		Select(pipelineVersionColumns...).
+		From("pipeline_versions").
+		Where(sq.Eq{"UUID": versionId}).
```

Comment:
can you use and statement instead of chaining where for better readability

Reply:
done